### ISSUE TITLE
Create a tiny version of flan-t5-small model

### ISSUE THREAD

**Issue by ndersam** [ZHCSJ666/ML-24-25#15]
This issue covers experimentation related to flan-t5. The objective is to develop a small version of flan-t5 that can perform git commit message generation. 

The original version of flan-t5-small has around 77million parameters which is deemed too big for our project.

**Comment by ndersam** [comment]
I've trained a smaller version of the model with 22 million parameters on commit-chronicle dataset (Golang only and "ADD" changes only). The model overfits on the train dataset achieving the following metrics:

| | | 
|---|---|
| train/MRR_top5 | 0.9260 |
| train/acc_top1 | 0.8833 |
| train/acc_top5 | 0.9827 |
| train/loss | 0.5432 |
| val/MRR_top5 | 0.2840 |
| val/acc_top1 | 0.2424 |
| val/acc_top5 | 0.3510 |
| val/loss | 9.1635|

This is for 200 epochs.

The changed parameters of the model are:
d_ff: 512
d_model: 256
num_decoder_layers: 4
num_heads: 4
num_layers: 4

The training is much faster than the original t5-model that took 1hr 20mins for one epoch. One epoch training this model on the dataset described above took less than 2 minutes.

**Comment by rissalMhedna** [comment]
It seems that the model is overfitting, which is why val/acc.. are so low. We may need early stopping or more dropout
